V0

firas-yangui opened the pull request:
i will create myMethods.js and exports function to clean up the code!

firas-yangui committed:
V0 

i will create myMethods.js and exports function to clean up the code!

firas-yangui committed:
Create ServerV1.js

firas-yangui committed:
Create ClientV1.js

firas-yangui committed:
Create myMethods.js

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+app.get('/bike/:id', function(req, res) {
+
+	myMethods.getBikeById(req.params.id,function(err,bike){
+		if(err)
```
C'est dommage d'avoir une erreur et de renvoyer 404. Une erreur ca doit retourner 500. Si jamais ton erreur c'est que tu ne trouves pas l'element, tu dois tester !bike et pas err.

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+		}
+		else
+		{
+		return res.status(200).json(myMethods.convertJsonFiel(bike)); // nous pouvons utilisé .send	
```
Typo dans convertJsonFiel non?

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+});
+
+//bikes with params
+app.get('/bikes/:country', function(req, res) {
```
Tu définis la même route qu'au dessus. Donc une va écraser l'autre.

chamerling replied:
J'avais demander d'utiliser un query parameter, donc req.query.country par exemple.

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+});
+
+
+app.delete('/bike/:id', function (req, res) {
```
On fait delete sur bikes et non sur bikes.

```
app.delete('/bikes/:id', function(req, res) {
  ...
});
```

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+	myMethods.getIndexBike(req.params.id,function(err,indexToRemove){
+		if (err)
+			{
+				res.status(404);
```
Même commentaire pour 404 et err que précédemment

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+			}
+			else
+			{
+				myMethods.bikes.splice(indexToRemove,1);
```
C'est bien de chercher puis supprimer.

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+			else
+			{
+				myMethods.bikes.splice(indexToRemove,1);
+				return res.status(202).send();//status accepted!
```
La suppression c'est 204, pas 202

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+});
+
+
+app.put('/bike', function(req, res) {
```
app.put('/bikes'

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+
+app.put('/bike', function(req, res) {
+	
+	myMethods.bikeExiste(req.body.id,function(err){
```
On ne mixe pas EN et FR.

chamerling commented on ServerV1.js:
```diff
@@ -0,0 +1,95 @@
+		}
+		else
+		{
+			var id = req.body.id;
```
On ne laisse jamais le client générer un id. C'est le role de la base ou du serveur.

chamerling commented on client.js:
```diff
@@ -1,6 +1,15 @@
+
+request({
+   method: 'PUT',
+   uri: 'http://localhost:3000/bike',
```
/bikes

chamerling commented on myMethods.js:
```diff
@@ -0,0 +1,101 @@
+];
+
+
+var getBikeById = function(id,callback) {
```
cf Array.find pour simplifier le code

chamerling commented on myMethods.js:
```diff
@@ -0,0 +1,101 @@
+
+var getBikeByCountry = function(country)	{
+	var tabRes=[];
+	for(var i=0;i<bikes.length;i++)
```
cf Array.find aussi

chamerling commented on myMethods.js:
```diff
@@ -0,0 +1,101 @@
+	var tabRes=[];
+	for(var i=0;i<bikes.length;i++)
+	{
+		if (bikes[i].country== country)
```
Attention a la casse, il faut pouvoir comparer US et us

chamerling commented on myMethods.js:
```diff
@@ -0,0 +1,101 @@
+};
+
+
+var getIndexBike = function(id,callback){
```
cf Array.findIndex

chamerling commented on myMethods.js:
```diff
@@ -0,0 +1,101 @@
+{
+	for(var i=0;i<this.bikes.length;i++)
+	{
+		if(bikes[i].id== id)
```
On aime pas comparer les strings et les nombres même si c'est possible en JS

chamerling commented on myMethods.js:
```diff
@@ -0,0 +1,101 @@
+	return callback(null);
+}
+
+function convertJsonsFiel(jsons)
```
Donc j'aurais toujours le prix en dollar et plus jamais en euro dans mon API?
Ca aurait été mieux d'utiliser un query parameter pour demander la devise de sortie:

?currency=us

firas-yangui replied:
je vais régler ce problème ce soir, normalement je vais chercher la piste d'utiliser les prototypes
(si vous avez un livre ou un tuto a me proposer ?)

chamerling replied:
Tu peux le faire sans prototype, keep it simple

firas-yangui replied:
j'ai modifié les fonctions de conversions pour créer un autre objet,
et j'ai fait l'appel en testant sur un query params ?price=us

chamerling commented:
Je ne comprends pas pourquoi j'ai des fichiers V1 au lieu d'avoir tout dans client et server.js

firas-yangui commented:
bonjour Christophe,

Merci pour ton retour,

j'ai essayé de faire une autre version V1 dans laquelle j'ai mis les
méthodes dans un fichier JS séparé,
je suis à Sofrecom, je corrige les erreurs ce soir,

Cordialement,

2016-04-25 15:32 GMT+01:00 Christophe Hamerling notifications@github.com:

> Je ne comprends pas pourquoi j'ai des fichiers V1 au lieu d'avoir tout
> dans client et server.js
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/chamerling/express-challenge/pull/5#issuecomment-214361911

## 

Firas YANGUI
Student, sesame university
Mobile: (+216) 20 123 886

firas-yangui committed:
aprés modification

j'ai pas modifié les conv du prix us->euro
je vais le faire demain soir

firas-yangui committed:
modification du conversion euro -->dollar

utilisation du création d'objet pour ne pas écraser le tableaux de bikes initiale,
le choix se fait selon un query params

chamerling commented on myMethods.js:
```diff
@@ -1,101 +1,119 @@
+
+var getBikeByCountry = function(country){
+	var tabRes=[];
+	tabRes = _.filter(bikes, function(obj) {
```
tu peux faire bikes.filter(function(bike)) {} et ne pas utiliser _

firas-yangui replied:
c'est fait, j'ai modifié le code

chamerling commented on myMethods.js:
```diff
@@ -1,101 +1,119 @@
+}
+
+
+function convertJsonFiel(json) //j'ai joué sur la création d'un nouveau obj pour que le tableau initiale ne change pas!
```
c'est inutile, regarde Array.map

firas-yangui committed:
Utilisation du Array.map & Array.filter
